embassy-stm32/src/rcc/g4.rs

```diff
@@ -262,20 +291,22 @@ pub(crate) unsafe fn init(config: Config) {
         }
     };
 
-    // Setup the 48 MHz clock if needed
-    if let Some(clock_48mhz_src) = config.clock_48mhz_src {
-        let source = match clock_48mhz_src {
-            Clock48MhzSrc::PllQ => {
+    // Configure the 48MHz clock source for USB and RNG peripherals.
+    {
+        let source = match config.clk48_src {
+            Clk48Src::PLL1_Q => {
                 // Make sure the PLLQ is enabled and running at 48Mhz
                 let pllq_freq = pll_freq.as_ref().and_then(|f| f.pll_q);
                 assert!(pllq_freq.is_some() && pllq_freq.unwrap().0 == 48_000_000);
```

Comment:
these asserts now mean it's now impossible to *not* have a 48mhz clock somewhere. IMO this should let through whatever invalid config (clock off or bad freq) and propagate it to `set_clocks!`. `l.rs` [does this.](https://github.com/embassy-rs/embassy/blob/main/embassy-stm32/src/rcc/l.rs#L270-L283).

Then we can add an assert to USB on startup, that fails if clock is off or wrong. This way it only fails if you actually want USB.

Reply:
Okay, and the same for the RNG peripheral which also uses the 48MHz clock?

Reply:
yeah. I think that one doesn't require exactly 48mhz though, or at least not in all families. 

By "Then we can add an assert to USB on startup" I mean later, not in this PR. For that to work the clock muxes for USB must be well implemented on all families, which they aren't right now :(

Reply:
Yep, I assumed you meant the USB clock assert would happen elsewhere. IIRC some stm32-rs HALs have some kind of `assert_usb_clock_valid!()` macro they use for this

Reply:
yeah, it'd be when you create a USB driver.